src/client/Microsoft.Identity.Client/ManagedIdentity/V2/MtlsBindingStore.cs

```
@@ -0,0 +1,183 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client.ManagedIdentity.V2
+{
+    /// <summary>
+    /// Centralized helper for installing, locating and pruning IMDSv2 client mTLS binding certs
+    /// from the CurrentUser\My store.
+    /// </summary>
+    internal static class MtlsBindingStore
+    {
+        // Minimum remaining lifetime required to reuse a binding.
+        internal static readonly TimeSpan MinFreshRemaining = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// Installs the certificate in CurrentUser\My, removes any exact-duplicate thumbprints,
+        /// and best-effort prunes older certs for the same subject. Returns the subject DN used
+        /// as lookup key.
+        /// </summary>
+        internal static string InstallAndGetSubject(X509Certificate2 cert, ILoggerAdapter logger = null)
+        {
+            if (cert == null)
+                return null;
+
+            using var store = new X509Store(StoreName.My, StoreLocation.CurrentUser);
+
+            try
+            {
+                store.Open(OpenFlags.ReadWrite);
+
+                // Avoid duplicates by thumbprint
+                foreach (var dup in store.Certificates.Find(X509FindType.FindByThumbprint, cert.Thumbprint, validOnly: false))
+                {
+                    try
+                    { store.Remove(dup); }
+                    catch { /* best effort */ }
+                }
+
+                store.Add(cert);
+
+                // Best effort prune for same subject (keep newly added one)
+                TryPruneOlderForSubject(store, cert.Subject, keepThumbprint: cert.Thumbprint, logger);
+            }
+            catch
+            {
+                // If store operations fail, carry on; the cert is still available in-memory for this call.
```

Avoid ignoring exceptions. If operation is non-critial, at least log?